contracts/core/Registry.sol

```diff
@@ -1,242 +1,110 @@
-// SPDX-License-Identifier: AGPL-3.0-only
-pragma solidity 0.8.19;
-
-import {Metadata} from "../core/libraries/Metadata.sol";
-import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
-
-contract Registry is Initializable {
-    // A linked list of owners of a identity
-    // The use of a linked list allows us to easily add and remove owners,
-    // access them directly in O(1), and loop through them.
-    //
-    // {
-    //     count: 3,
-    //     list: {
-    //         OWNERS_LIST_SENTINEL => owner1Address,
-    //         owner1Address => owner2Address,
-    //         owner2Address => owner3Address,
-    //         owner3Address => OWNERS_LIST_SENTINEL
-    //     }
-    // }
-    struct OwnerList {
-        uint256 count;
-        mapping(uint => address) list;
-    }
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { OwnableRoles } from "../../lib/solady/src/auth/OwnableRoles.sol";
+import { Metadata } from "./libraries/Metadata.sol";
+
+// use Solady Roles for ownership of projects: https://github.com/Vectorized/solady/blob/main/src/auth/OwnableRoles.sol
+contract Registry is OwnableRoles {
+
+    uint private identityId;
 
     struct IdentityDetails {
-        uint id;
         string name;
-        Metadata.MetaPtr metadata;
-        address attestationAddress;
+        Metadata metadata;
+        address attestationAddr;
+        // address coreOwner; // ASK: Should this be added?
```

Comment:
Since this contract uses Roles, we can use solady to track owners but how do we track coreOwner without capturing it separately ?

Reply:
Do we want coreOwner? I thought on the call yesterday we decided we were fine without the ability to look up owner, because the legible version of who they are is captured in the name and metadata, so owner would just be an address that wouldn't add much. 

If we decide it's needed, agreed it should be added to this struct.